packages/addons-v5/lib/addons_wait.js

```diff
@@ -1,26 +1,60 @@
+/* eslint-disable no-await-in-loop */
+
 'use strict'
 
 const cli = require('heroku-cli-util')
 
-module.exports = async function (api, addon, interval) {
-  const app = addon.app.name
-  const addonName = addon.name
+module.exports = {
+  waitForAddonProvisioning: async function (api, addon, interval) {
+    const app = addon.app.name
+    const addonName = addon.name
+
+    await cli.action(`Creating ${cli.color.addon(addon.name)}`, (async function () {
+      while (addon.state === 'provisioning') {
+        // eslint-disable-next-line no-promise-executor-return
+        await new Promise(resolve => setTimeout(resolve, interval * 1000))
+
+        addon = await api.request({
```

Comment:
Please create a new var instead of overwriting `addon`.

Reply:
Here is difficult to use a different var, because this var is used to test for the `while` loop exit condition. If we don't reassign to the same var, it will never end.

Reply:
I still find this pretty uncomfortable. 
You could do this in all the places I've left a comment:
`let addonResponse = {...addon}`
and
`addonResponse = await api.request({`

Reply:
Fixed following your great suggestion.